docs/dotnet/dotnet-http-metrics.md

```diff
@@ -75,23 +73,20 @@ Notes:
 |---|---|
 | `active` | active state. |
 | `idle` | idle state. |
-<!-- endsemconv -->
 
 ### Metric: `http.client.connection.duration`
 
 this metric SHOULD be specified with
 [`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advisory-parameters)
 of `[ 0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30, 60, 120, 300 ]`.
 
-<!-- semconv metric.dotnet.http.client.connection.duration(metric_table) -->
+<!-- Tables in this document are not auto-generated and are intentionally frozen in time. From the .NET perspective this metric and its attributes are stable till the next major version. They are still experimental in the OpenTelemetry. -->
 | Name     | Instrument Type | Unit (UCUM) | Description    |
 | -------- | --------------- | ----------- | -------------- |
 | `http.client.connection.duration` | Histogram | `s` | The duration of the successfully established outbound HTTP connections. [1] |
```

Comment:
@antonfirsov I noticed we don't have `error.type` on this metric (also in the [code](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/ConnectionMetrics.cs#L72)), I don't remember if there were some blockers on adding it. 

Do you remember any details? How do we know if connection finished gracefully or was terminated?

Reply:
It's very untrivial what exact error information should we log here. We don't do it in our connection telemetry and would be hesitant to introduce anything without a specific user request driven by specific scenarios. Even if there would be such an ask, it wouldn't be easy to plumb it through the implementation.

Reply:
I see where you're coming from. I understand it could be hard to get this information, but I want to challenge the assumption that it's not useful.

Today .NET version of `http.client.request.duration` captures time-to-headers

https://github.com/open-telemetry/semantic-conventions/blob/c8b61636277c9577a8e1948f1df70b4da8716d2d/docs/dotnet/dotnet-http-metrics.md?plain=1#L31

so users never see connection terminations that happen when reading response streams in their metrics (when not buffered) OR even traces.

As a former service developer, I think it's very valuable to know the rate of transient connection resets/timeouts. Or if I have a peak of such errors. I'd definitely wanted to know that connection ended gracefully or was terminated with an error. Http clients allow to configure read/write timeouts - they'd probably want to have metrics to tune this parameters.

Unrelated to HTTP, but in client libraries that use AMQP we don't have connection pooling and connection termination reason is extremely important (at least if it's graceful or not).

What I'm saying here is that the future generic version of `http.client.connection.duration` (or `connection.client.duration`) has a very high chance of getting `error.type` attribute (as any other `.duration` metric)

Reply:
Created #809 to track it